Assets/Scripts/GameCharacter/GameCharacterAnimationController.cs

```diff
@@ -1,70 +0,0 @@
-using DG.Tweening;
-using System;
-using System.Collections;
-using System.Collections.Generic;
-using TMPro;
-using UnityEngine;
-
-public class GameCharacterAnimationController : GameCharacterComponent
-{
-
-    [SerializeField] TextMeshPro statusDisplay;
-    [SerializeField] MeshRenderer gunBarrelMeshRendered;
-    [SerializeField] MeshRenderer characterBodyMeshRendered;
-    [SerializeField] GameObject gunSmokeParticalEffect;
-    [SerializeField] Color attackAnimationColor, takeDamageAnimationColor;
-
-    GameCharacterController characterController;
```

Comment:
Your dependencies should be interfaces.
You can even expose them in the inspector with a trick (this is one of the solutions out there https://www.patrykgalach.com/2020/01/27/assigning-interface-in-unity-inspector/)

Of course you can avoid that when using Unity types